tests/test_greet.py

```diff
@@ -15,3 +15,17 @@ def test_greet_without_name():
     result = greet(data)
     expected = "Hello, my friend."
     assert result == expected
+
+
+def test_greet_with_uppercase_name():
+    data = "JERRY"
+    result = greet(data)
+    expected = "HELLO {}!".format(data)
+    assert result == expected
+
+
+def test_greet_with_two_names_from_list():
+    data = ["Jill", "Jane"]
```

Comment:
this is a good usecase for pytest parametrize feature https://docs.pytest.org/en/stable/parametrize.html let's use it.